pkg/http/server.go

```diff
@@ -72,7 +74,14 @@ func (s *server) setupListener(opts Opts) error {
 	networkType := getNetworkScheme(opts.BindAddress)
 	listenAddr := getListenAddress(opts.BindAddress)
 
-	listener, err := net.Listen(networkType, listenAddr)
+	listener, err := func() (net.Listener, error) {
+		if networkType == "unix" {
+			return setupUnixSocketListener(networkType, listenAddr)
+		} else {
+			return net.Listen(networkType, listenAddr)
```

Comment:
Why not do like below?
```go
listener, err := net.Listen(networkType, listenAddr)
if err != nil {
  return err
}
if networkType == "unix" {
  // chmod here
}
```

Reply:
Ah, good point. Artifact of an intermediate code version where I was trying to use FileListener (before learning that you can't use it with sockets), so it had to be split at the time.

Will replace it as you suggest :+1:

Reply:
Unfortunately it doesn't actually work, because the plain
```go
net.Listen(..., listenAddr)
```

will turn out to be
```go
net.Listen(..., "/path/to/foo.sock,mode=...")
```

And thus create a socket whose name includes the parameters...

Looking into ways to address this